Packages/UIComponents/Sources/UIComponents/Models/AlertPresentationType.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  File.swift
+//  
+//
+//  Created by Eric Silverberg on 12/21/22.
+//
+
+import Foundation
+
+public enum DeepLinkDestination {
+    case none
+}
+
+public enum AlertPresentationModality {
+    case none
+    case dialog
+    case toast
+}
+
+public protocol AlertPresentationTypeProviding {
+    var alertPresentation: AlertPresentationType { get }
+}
+
+public struct AlertPresentationType {
```

Comment:
Having Type on this name confuses me a bit.
This class looks to me more like a factory, I think we already have AlertPresentationModality, which is an enum, of toast and dialog. 
I think we could have something like AlertPresentationFactory, that would build an AlertPresentation based on AlertPresentationModality.